explanation/dnssec/dnssec.md

```diff
@@ -0,0 +1,254 @@
+
+
+## Where does the DNSSEC validation happen?
+
+DNSSEC validation is the act of fetching the DNS data that was requested, their signatures, and performing the cryptographic validation of that data. Who is responsible for that?
+
+It depends.
+
+Let's analyze the simple scenario of a system on a local network performing a DNS query for a domain.
+
+![Simple DNS](../images/ubuntu-local-recursive-dns.png)
+
+Here we have:
+
+ * An Ubuntu system, like a desktop, configured to use a DNS server in the local network.
+ * A DNS server configured to perform recursive queries on behalf of the clients from the local network.
+
+Let's zoom in a little bit on that Ubuntu system:
+
+![Stub Resolver](../images/ubuntu-stub-resolver.png)
+
+When an application needs to translate a hostname to an IP address, it uses standard glibc calls for that job. That is called a stub resolver, or simply a "dns client". This is a very simple client in the sense that it does not perform recursive queries: it expects to dispatch the DNS query to a recursive DNS server, which will do all the hard work.
+
+In Ubuntu, the default stub resolver is `systemd-resolved`. That's a daemon, running locally, and listening on port 53/udp on IP 127.0.0.53. The system is configured to use that as its nameserver via `/etc/resolv.conf`:
+
+    nameserver 127.0.0.53
+    options edns0 trust-ad
+
+This stub resolver has its own configuration for which recursive DNS servers to use. That can be seen with the command `resolvectl`. For example:
+
+    Global
+             Protocols: -LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
+      resolv.conf mode: stub
+
+    Link 12 (eth0)
+        Current Scopes: DNS
+             Protocols: +DefaultRoute -LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
+    Current DNS Server: 10.10.17.1
+           DNS Servers: 10.10.17.1
+            DNS Domain: lxd
+
+This configuration is usually provided via DHCP, but could also be set via other means. In this particular example, the DNS server that the stub resolver (`systemd-resolved`) will use for all queries that go out on that network interface is 10.10.17.1. The output above also has `DNSSEC=no/unsupported`: we will get back to that in a moment, but it means that `systemd-resolved` is not performing the DNSSEC cryptographic validation.
+
+Given what we have:
+ * an application
+ * stub resolver ("dns client")
+ * recursive DNS server in the local network
+ * several other DNS servers in the internet that will be queried by our recursive DNS server
+
+Where does the DNSSEC validation happen? Who is responsible?
+
+Well, any DNS server can perform the validation. Let's look at two scenarios, and what it means in each case.
+
+### Validating Resolver
+
+When a recursive DNS server is also performing DNSSEC validation, it's called a *Validating Resolver*. That will typically be the DNS server on your local network, at your company, or in some cases at your ISP.
```

Comment:
I notice that while here you capitalize "Validating Resolver" consistently in this document, in dnssec-troubleshooting.md and install-dnssec.md you have it lower case.  I think it would be best to have it capitalized throughout, for overall consistency.

Reply:
indeed, replaced all over.